src/process_utility.c

```diff
@@ -536,6 +630,58 @@ process_altertable(Node *parsetree)
 	cache_release(hcache);
 }
 
+typedef struct CreateIndexCtx
+{
+	CreateTrigStmt *stmt;
+	const char *queryString;
+	Oid			hypertable_relid;
+} CreateIndexCtx;
+
+static void
+create_trigger_chunk(Oid chunk_relid, void *arg)
+{
+	CreateIndexCtx *ctx = (CreateIndexCtx *) arg;
+	CreateTrigStmt *stmt = ctx->stmt;
+
+	Oid			trigger_oid = get_trigger_oid(ctx->hypertable_relid, stmt->trigname, false);
+
+	char	   *relschema = get_namespace_name(get_rel_namespace(chunk_relid));
+	char	   *relname = get_rel_name(chunk_relid);
+
+	trigger_create_on_chunk(trigger_oid, relschema, relname);
+}
+
+static void
+process_create_trigger(CreateTrigStmt *stmt, const char *query_string)
+{
+	Cache	   *hcache = NULL;
+	Hypertable *ht;
+	Oid			relid = RangeVarGetRelid(stmt->relation, NoLock, true);
+	CreateIndexCtx ctx = {
+		.stmt = stmt,
+		.queryString = query_string,
+		.hypertable_relid = relid
+	};
+
+	if (!stmt->row)
+		return;
+
+	if (!OidIsValid(relid))
+		return;
+
+
+	hcache = hypertable_cache_pin();
+
+	ht = hypertable_cache_get_entry(hcache, relid);
```

Comment:
There is a convenience function `is_hypertable()` that you can use if you only need to know that the relation is a hypertable and you don't need the hypertable info. That allows you to get rid of the cache pinning as well.